jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,427 @@
+                    .setQuery(QueryBuilders.existsQuery(fieldToAdd))
+                    .get();
+            String script;
+            if(existsResponse != null && existsResponse.getHits() != null && existsResponse.getHits().totalHits() > 0) {
+                //This means field already exists and therefore we should append to it
+                script = "ctx._source." + fieldToAdd+".add('"+ fieldValueToAdd + "')";
+            } else {
+                //The field does not exists. so we create one
+                script = "ctx._source." + fieldToAdd+" =['"+ fieldValueToAdd + "']";
+            }
+
+
+
+            UpdateRequest upReq = new UpdateRequest(INDEX_NAME, docDef.getEntityField(), entity.getId())
+                    .script(new Script(script))
+                    .upsert(indexRequest);
+
+            UpdateResponse response = client.update(upReq).get();
+
+            LOGGER.debug("Received the following Update response : " + response + " for the following entity: " + entity);
+
+        } catch(Exception e) {
+            throw new TextIndexException("Unable to Index the Entity in ElasticSearch.", e);
+        }
+
+
+    }
+
+    /**
+     * Delete an entity.
+     * Since we are storing different predicate values within the same indexed document,
+     * deleting the document using entity Id is sufficient to delete all the related contents for a given entity.
+     * @param entity entity to delete
+     */
+    @Override
+    public void deleteEntity(Entity entity) {
+
+        String fieldToRemove = null;
+        String valueToRemove = null;
+        for(String field : docDef.fields()) {
+            if(entity.get(field) != null) {
+                fieldToRemove = field;
+                valueToRemove = (String)entity.get(field);
+                break;
+            }
+        }
+        //First Search of the field exists or not
+        SearchResponse existsResponse = client.prepareSearch(INDEX_NAME)
+                .setTypes(docDef.getEntityField())
+                .setQuery(QueryBuilders.existsQuery(fieldToRemove))
+                .get();
+
+        String script = null;
+        if(existsResponse != null && existsResponse.getHits() != null && existsResponse.getHits().totalHits() > 0) {
+            //This means field already exists and therefore we should remove from it
+            script = "ctx._source." + fieldToRemove+".remove('"+ valueToRemove + "')";
+        }
+
+        UpdateRequest updateRequest = new UpdateRequest(INDEX_NAME, docDef.getEntityField(), entity.getId())
+                .script(new Script(script));
+
+        try {
+            client.update(updateRequest).get();
+        }catch(Exception e) {
+            throw new TextIndexException("Unable to delete entity.", e);
+        }
+
+
+        LOGGER.debug("deleting content related to entity: " + entity.getId());
+//        client.prepareDelete(INDEX_NAME, docDef.getEntityField(), entity.getId()).get();
```

Comment:
Please remove commented out lines like this, unless they have a clear purpose and explanation